config/peerpods/podvm/gcp-podvm-image-handler.sh

```diff
@@ -188,6 +202,44 @@ function create_image_from_prebuilt_artifact() {
   echo "GCP image created successfully from prebuilt artifact"
 }
 
+function image_exists() {
+    echo "Checking if GCP image exists"
+
+    local image_path latest_image_id
+
+    # This command returns something like this:
+    # https://www.googleapis.com/compute/v1/projects/my-project/global/images/my-image
+    # Use `cut` to the image name
+    image_path=$(gcloud compute images describe "${IMAGE_NAME}-${IMAGE_VERSION}" \
+        --project="${GCP_PROJECT_ID}" \
+        --format='get(selfLink)' 2>/dev/null | cut -d/ -f6-)
+
+    if [[ $? -ne 0 ]]; then
+        echo "Failed to get image ${IMAGE_NAME}-${IMAGE_VERSION} from GCP."
+        return 1
+    fi
+
+    latest_image_id=$(kubectl get configmap peer-pods-cm \
+        -n openshift-sandboxed-containers-operator \
+        -o jsonpath='{.metadata.annotations.LATEST_IMAGE_ID}')
+
+    if [[ -z "${image_path}" && -z "${latest_image_id}" ]]; then
+        echo "No image in GCP and no record in configmap."
+        return 1
+    elif [[ -z "${image_path}" && -n "${latest_image_id}" ]]; then
+        echo "No image in GCP, but configmap has record (${latest_image_id})."
+        return 1
+    # Since GCP supports both cases, lets see if image_path is a suffix of
+    # latest_image_id.
+    elif [[ "${image_path}" == *"${latest_image_id}" ]]; then
+        echo "Image (${latest_image_id}) is up-to-date in configmap."
+        return 0
+    else
+        echo "Image mismatch: GCP image (${image_path}) different from ConfigMap (${latest_image_id})."
+        return 2
+    fi
```

Comment:
This `image_exists` function turns out to have non-trivial semantics, not just returning true or false. IMO it would be clearer to open-code it in its unique caller. This would spare the need for the exit status values.

Reply:
> gcp: adding image check before creating image
> As following other providers logic.
>
> Signed-off-by: Beraldo Leal <bleal@redhat.com>

The commit message would be a good place to describe the various cases that need to be handled BTW.

In the meantime, I went to #584 and I realized that you borrowed the convoluted `image_exists` pattern from there and it is already merged :disappointed: ... well, I guess I won't fight if everyone is apparently happy with that.